far/FarEng.hlf.m4

```diff
@@ -4225,54 +4225,53 @@ and the '|' character separates include masks from exclude masks.
-одной или нескольких разделённых запятыми ~масок файлов~@FileMasks@. #Gray *#
-инвертирует текущую пометку. Если параметр #Пометка папок# в диалоге ~Настроек панели~@PanelSettings@
-включен, то пометка меняется и для папок.
+    #Shift+Arrow keys# move the cursor while selecting or deselecting
+items along the way. The action (selection or deselection) depends
+on the state of the item under cursor before pressing the key
+combination.
 
-    #Ctrl+<Gray +># и #Ctrl+<Gray -># выбирают и снимают пометку со всех
-файлов с тем же расширением, что и у файла под курсором. Если параметр
-#Пометка папок# в диалоге ~Настроек панели~@PanelSettings@ включен, то
-пометка меняется и для папок.
+    #Gray +# selects, and #Gray -# deselects the items using one or more
+~File masks~@FileMasks@. #†#
 
-    #Alt+<Gray +># и #Alt+<Gray -># выбирают и снимают пометку со всех файлов
-с тем же именем, что и у файла под курсором. Если параметр #Пометка папок# в диалоге
-~Настроек панели~@PanelSettings@ включен, то пометка меняется и для папок.
+    #Gray *# inverts the current selection. #†#
 
-    #Gray *# инвертирует текущую пометку для файлов, или для файлов и папок,
-если параметр #Пометка папок# в диалоге ~Настроек панели~@PanelSettings@ включен.
+    #Ctrl+<Gray +># selects, and #Ctrl+<Gray -># deselects all items
+with the same #extension# as that of the item under cursor. #†#
 
-    #Ctrl+<Gray *># инвертирует текущую пометку, включая папки.
+    #Alt+<Gray +># selects, and #Alt+<Gray -># deselects all items with
+the same #name# as that of the item under cursor. #†#
 
-    #Shift+<Gray +># и #Shift+<Gray -># выбирают и снимают пометку со всех
-файлов, или для файлов и папок, если параметр #Пометка папок# в диалоге
-~Настроек панели~@PanelSettings@ включен.
+    #Shift+<Gray +># selects, and #Shift+<Gray -># deselects all items. #†#
 
-    #Ctrl+M# (#Восстановить пометку#) восстанавливает предыдущую
-выделенную группу.
+    #†# If the #Select folders# option of the
+~Panel settings~@PanelSettings@ dialog is off, only files are selected
+or deselected. Otherwise, the selection on the folders is changed
+as well.
 
+    #Ctrl+<Gray *># inverts the current selection on all items,
+including folders.
 
- #Пометка мышью#
+    #Ctrl+M# restores the last selection.
 
-    #Щелчок правой кнопкой мыши# помечает или снимает пометку с файла или
-папки под курсором мыши.
+  #Mouse Selection#
 
-    Если щелкнуть правой кнопкой мыши над списком файлов (строка
-с заголовками колонок) или под списком файлов (статусная область) то при
-удерживании правой кнопкой мыши список файлов будет прокручиваться вверх
-или вниз помечая или снимая пометку с файлов, по которым пробегает курсор.
-Действие (пометка или снятие пометки) соответствует состоянию файла под
-курсором до начала прокрутки.
+    #Right click# toggles selection on the clicked item and moves the
+cursor to it.
 
+    #Right click and hold# on the #status line# or #column titles# moves
```

Comment:
1. It looks like this feature heavily glitches. It seems to take into account the state of the next item, not the item under cursor. Also, the behavior somehow depends on the current state of the selection, if not on the history of the cursor movement. I could not formulate the exact rules.
2. The cursor moves way too fast, even with debug build. It is difficult or impossible to use this feature meaningfully.

Reply:
1 - what do you mean by "take into account"? It just scrolls.
2 - yes, we should probably add some sleep() there to emulate keyboard speed (same for editor/viewer).

Reply:
2. Is `sleep()` appropriate? It is blocking. The "right" solution would be to schedule a timer event and exit the mouse handler. Is such technique used elsewhere in Far? I fully understand that it's much more involved. So, if `sleep()` is OK, especially if it is used in other places, I'll try to apply it here.

Reply:
It is ok, used in other places and used to be here too I think.
Probably somewhere around or in `IsMouseButtonPressed()` function.